transfers/amclient.py

```diff
@@ -0,0 +1,518 @@
+                r = self.hide_transfer(transfer_uuid)
+                if r.status_code == 200:
+                    ret.setdefault('close_succeeded', []).append(transfer_uuid)
+                    LOGGER.info('Closed transfer {0}.'.format(transfer_uuid))
+                else:
+                    ret.setdefault('close_failed', []).append(transfer_uuid)
+                    LOGGER.warning('FAILED to close transfer {0}.'
+                                   .format(transfer_uuid))
+        return ret
+
+    def completed_transfers(self):
+        """Return all completed transfers. GET /transfer/completed::
+
+            $ ./amclient.py completed-transfers \
+                --user=test \
+                --api-key=e8f8a0fb157f08a260045f805455e144d8ad0a5b
+
+        """
+        return self._call_url_json(
+            '{}/api/transfer/completed'.format(self.am_url), self._am_auth())
+
+    def unapproved_transfers(self):
+        """Return all unapproved transfers. GET transfer/unapproved::
+
+            $ ./amclient.py unapproved-transfers \
+                --user=test \
+                --api-key=e8f8a0fb157f08a260045f805455e144d8ad0a5b
+
+        """
+        return self._call_url_json(
+            '{}/api/transfer/unapproved'.format(self.am_url), self._am_auth())
+
+    def transferables(self, b64decode=False):
+        """Return all transferable entities in the Storage Service.
+        GET location/<TS_LOC_UUID>/browse/::
+
+            $ ./amclient.py transferables \
+                --ss-user=test \
+                --ss-api-key=7558e7485cf8f20aadbd95f6add8b429ba11cd2b \
+                --transfer-source=7ea1eb0e-5f4e-42e0-836d-c9b4ab5692e1 \
+                --transfer-path=vagrant/archivematica-sampledata
+
+        """
+        url = '{}/api/v2/location/{}/browse/'.format(
+            self.ss_url, self.ts_location_uuid)
+        params = self._ss_auth()
+        if self.ts_path_prefix:
+            params['path'] = base64.b64encode(self.ts_path_prefix)
+        result = self._call_url_json(url, params)
+        if b64decode:
+            return self.b64decode_ts_location_browse(result)
+        return result
+
+    def b64decode_ts_location_browse(self, result):
+        """Base64-decode the results of a call to SS GET
+        /location/UUID/browse/.
+        """
+        def dec(thing):
+            return base64.b64decode(getattr(thing, 'encode')('utf8'))
+        try:
+            result['directories'] = [dec(d) for d in result['directories']]
+            result['entries'] = [dec(e) for e in result['entries']]
+            result['properties'] = dict((dec(k), v) for k, v in
+                                        result['properties'].items())
+            return result
+        except Exception:
+            return result
+
+    def search_package(self, params):
+        """SS GET /search/package?<GET_PARAMS>."""
+        return self._call_url_json(
+            '{}/api/v2/search/package/'.format(self.ss_url), params)
```

Comment:
Resources in the storage service already allow you to filter by certain fields. Can this be done using those instead of requiring the new /search/ endpoint? eg /api/v2/file/?package_type=AIP